napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
 
 regionprops_table_all_frames = analyze_all_frames(regionprops_table)
 register_function(regionprops_table_all_frames, menu="Measurement > Regionprops of all frames (nsr)")
+
+
+def make_element_wise_dict(list_of_keys, list_of_values):
+    """Return an element-wise dictionary of two lists.
+
+    From a list of keys and a list of values, return a dictionnary linking
+    those lists. Both lists must have the same length.
+
+    Parameters
+    ----------
+    list_of_keys : List
+        list to be used as dictionnary keys.
+    list_of_values : List
+        list to be used as dictionnary values.
+
+    Returns
+    -------
+    Dictionnary
+    """
+    return dict(
+                list(
+                    map(
+                        lambda list_of_keys, list_of_values:
+                        [list_of_keys, list_of_values],
+                        *[list_of_keys, list_of_values]
+                        )
+                    )
+                )
+
+
+@register_function(
+    menu="Measurement > Regionprops map multichannel (scikit-image, nsr)")
+def napari_regionprops_map_channels_table(
+        label_images: List[napari.types.LabelsData],
+        intensity_images: List[napari.types.ImageData],
+        reference_label_image: napari.types.LabelsData,
+        intersection_area_over_object_area: float = 0.5,
+        return_summary_statistics: bool = True,
+        size: bool = True,
+        intensity: bool = True,
+        perimeter: bool = False,
+        shape: bool = False,
+        position: bool = False,
+        moments: bool = False,
+        napari_viewer: Viewer = None) -> "pandas.DataFrame":
+    """
+    Add a table widget to a napari viewer with mapped summary statistics.
+
+    Adds a table widget to a given napari viewer with summary statistics that
+    relates objects of one reference label image to objects in other label
+    images. Matched intensity images should be given for intensity
+    measurements. If no intensity images are given, calculates only properties
+    not related to intensities. If a single label image is given, it executes
+    regular 'regionprops_table' function.
+    """
+    multichannel = True
+    suffixes = []
+    if napari_viewer is not None:
+        # store list of labels layers for saving results later
+        labels_layer_list = [None]*len(label_images)
+        for layer in napari_viewer.layers:
+            if type(layer) is napari.layers.Labels:
+                # Store in the same order as label_images
```

Comment:
From the description of the functionality above I do not expect anything saved in all those layers. As far as I understood the functionality, the `reference_label_image` is the thing we measure.

Reply:
This was moved to _multichannel.py and adapted there.
I iterate over napari labels layers and select the one provided to `label_image_reference` to store the resulting table. It also uses this layer name as a suffix for the table.